XamarinCommunityToolkit/Helpers/WeakEventManager.shared.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+using static System.String;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+	// Copy from https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Core/WeakEventManager.cs
+	// TODO: Remove when PR complete: https://github.com/xamarin/Xamarin.Forms/pull/12063
+	class WeakEventManager
+	{
+		readonly Dictionary<string, List<Subscription>> eventHandlers = new Dictionary<string, List<Subscription>>();
+
+		public void AddEventHandler<TEventArgs>(EventHandler<TEventArgs> handler,
+			[CallerMemberName] string eventName = null)
+			where TEventArgs : EventArgs
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			AddEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void AddEventHandler(EventHandler handler, [CallerMemberName] string eventName = null)
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			AddEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void HandleEvent(object sender, object args, string eventName)
+		{
+			var toRaise = new List<(object subscriber, MethodInfo handler)>();
+			var toRemove = new List<Subscription>();
+
+			if (eventHandlers.TryGetValue(eventName, out var target))
+			{
+				for (var i = 0; i < target.Count; i++)
+				{
+					var subscription = target[i];
+					var isStatic = subscription.Subscriber == null;
+					if (isStatic)
+					{
+						// For a static method, we'll just pass null as the first parameter of MethodInfo.Invoke
+						toRaise.Add((null, subscription.Handler));
+						continue;
+					}
+
+					var subscriber = subscription.Subscriber.Target;
+
+					if (subscriber == null)
+						// The subscriber was collected, so there's no need to keep this subscription around
+						toRemove.Add(subscription);
+					else
+						toRaise.Add((subscriber, subscription.Handler));
+				}
+
+				for (var i = 0; i < toRemove.Count; i++)
+				{
+					var subscription = toRemove[i];
+					target.Remove(subscription);
+				}
+			}
+
+			for (var i = 0; i < toRaise.Count; i++)
+			{
+				(var subscriber, var handler) = toRaise[i];
+				handler.Invoke(subscriber, new[] {sender, args});
+			}
+		}
+
+		public void RemoveEventHandler<TEventArgs>(EventHandler<TEventArgs> handler,
```

Comment:
[Here](https://github.com/xamarin/XamarinCommunityToolkit/pull/279/files#diff-7b0d6956e569148d639df133f8874a4aR58), when you break the parameters in different lines, you do it differently, can you standardize?